### DISCUSSION TITLE
Extremely noisy pot ADC readings

### DISCUSSION THREAD

**Discussion by ramiabraham** [ElectroTechnique/electrotechnique.github.io#17] [Software Issues]
I've been having this issue for a while, and I think I've tried everything I can by now, so thought I'd ask for help/advice. I'm getting very noisy pot ADC readings on all pots, which causes the display to rapidly cycle between all parameters several times per second, making the display unusable, since it's triggered on change of a potentiometer value.

- Firmware version: 2.10
- PCB rev: 1.3
- Tindie order provided ST7735 display
- 33x Bourns 10k pots

This occurs whether powering from an AC-powered device, or direct DC battery power (via powerbank).

I thought I might have a bad PCB - rare but possible - so I also breadboarded everything with different pots and two 4067 mux breakout boards, but get the same issue.

I also tried using a Teensy 3.6 and the 3.6 firmware, but still get the same issue.

Has anyone else encountered this? I was surprised to find nothing about it on Github, Tindie, or the PJRC forum.

Thanks - any help/direction on this would be appreciated, looking forward to using TSynth.

**Comment by Electrotechnique** [comment]
Hello, I can't think of a cause for this, except soldering or the wrong pots. Can you check the pots with a multimeter - they are 10K not 1M?

**Reply by ramiabraham** [reply]
Thank you for the reply - I did check the pots with multimeters as noted to confirm their voltage range; they're standard B103 and are the same dimensions as the Bourns PTV09, but manufactured by TT Electronics.

I've used them in several other projects, but they are cheaper than the Bourns pots - so I have some of the recommended pots on the way just in case, which should arrive in a few days, provided the US Postal Service continues to exist.

I also did some continuity testing on copper fills and various traces, and it appears the copper fill surrounding the pots is not connected to any GND on the Teensy 4.1. So I figured I'd scratch off some of the mask, and run a jumper from the pots copper fill GND to a known, connected GND elsewhere. 

I selected one of the GND pins on the tactile switches for this. Unfortunately for me, connecting that GND to the GND copper fill surrounding the pots results in the T4.1 not powering on when connected to power (more specifically, the display doesn't turn on; have not yet tested further). It's a strange problem, as I'm aware that a design error like this is very unlikely to exist on a board that's in production and owned by several others, so I'm not certain yet that this is the root cause of the issue.

I'll continue testing tomorrow regarding getting the pots connected to GND successfully, and will update if there's any success there. I'll also update whether or not the Bourns PTV09 pots resolve the issue, in case the problem pops up for anyone else.

**Reply by Electrotechnique** [reply]
Can you send me a high resolution picture of the PCB please. The GND for the analogue part is separate, but there is a trace that connects to the rest of the GND. JLCPCB electrically check the PCBs - or so they claim.  It sounds like a soldering problem somewhere actually.

**Reply by ramiabraham** [reply]
Thanks - just sent over via email (but can post here if desired)

**Reply by Electrotechnique** [reply]
I've received nothing, send it to [info@electrotechnique.cc](info@electrotechnique.cc) please.

**Reply by ramiabraham** [reply]
Just re-sent - spam possibly?

**Reply by ramiabraham** [reply]
*Sent fwd from non-gmail account as well just in case

**Comment by Electrotechnique** [comment]
Ok, yes they were in spam, which was unexpected. I can't see anything obviously odd. You're saying that you get the same problems when bread-boarding and using a different Teensy? If so, what is the common component that could be causing the problems?

**Reply by ramiabraham** [reply]
The only things I can think of are that I've used cheaper pots in each build. For the pcb, I used [P090 by TT Electronics](https://www.digikey.com/en/products/detail/tt-electronics-bi/P090S-14T20BR10K/2408854), for the breadboard version, I used [P160, larger typical panel-mount pots, also by TT](https://www.digikey.com/en/products/detail/tt-electronics-bi/P160KN-0QC15B10K/2408878) 

- [datasheet for pcb mount](https://www.ttelectronics.com/TTElectronics/media/ProductFiles/Potentiometers/Datasheets/P090.pdf)
- [datasheet for panel-mount](https://www.ttelectronics.com/TTElectronics/media/ProductFiles/Potentiometers/Datasheets/P160.pdf)

DigiKey is experiencing some huge delays currently, so I haven't yet received the Bourns PTV09 shipment, but will test/update once received. Will poke around more until then.

**Reply by Electrotechnique** [reply]
The pots are probably all fine, as long as they're 10K. A strange problem, but I suspect it's a soldering one. Check voltages!

**Reply by ramiabraham** [reply]
Confirmed voltages again (appears all good), and am using the same solder I'd use in any other project ([Kester leaded; Amazon link).](https://www.amazon.com/gp/product/B0149K4JTY/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1).

How can I confirm that the pots are actually connected to the AGND in use? I checked the V1.3 schematic ([confirming file I'm referencing](https://github.com/ElectroTechnique/TSynth-Teensy4.1/blob/master/Mechanical/TSynth%20Schematic%20V1.3.pdf))  and "_Used as Analogue GND_" I see to the right of what is _AGND_ in Teensy 3.6, but is just part of the DGND net in Teensy 4.1. None of the pots are connected to this pin or any other GND on the board. Thanks for the help - feel like I'm missing something obvious here.

**Comment by Electrotechnique** [comment]
The analogue ground is on pin G top right of the Teensy 4.1. It will connect to ground when the Teensy is installed.
![Capture](https://user-images.githubusercontent.com/50934180/110868779-89391d80-830c-11eb-87f4-133d3b2357d5.JPG)

**Reply by ramiabraham** [reply]
Okay thanks, that's what I figured. I'm thinking I might have a damaged Teensy 4.1, seems to be the only remaining practical explanation. It's my only one, so I just ordered a couple more from PJRC (will re-test and update thread then, along with new pots if they change anything).

I also did a reduced test of just a Serial print loop of all the analog values on both muxes, and knob turns don't affect any of the values - it seems like all the pins are floating (they're all giving voltage at roughly mid-range). Could be another indication of a damaged Teensy.

**Reply by ramiabraham** [reply]
Just connected a new T4.1 and the issue is _entirely_ gone. Thanks for checking through everything else with me. I've never encountered this issue - unstable ADC, even with hysteresis - due to apparent microcontroller board damage.  No visual indication of damaged area; also tested for shorts across pins but couldn't find anything.
